src/radical/pilot/agent/scheduler/spark.py

```diff
@@ -0,0 +1,97 @@
+
+__copyright__ = "Copyright 2013-2016, http://radical.rutgers.edu"
+__license__   = "MIT"
+
+
+import json
+import urllib2 as ul
+
+import radical.utils as ru
+
+from ... import utils     as rpu
+from ... import states    as rps
+from ... import constants as rpc
+
+from .base import AgentSchedulingComponent
+
+
+#===============================================================================
+#
+class Spark(AgentSchedulingComponent):
+
+    # FIXME: clarify what can be overloaded by Scheduler classes
+
+    # --------------------------------------------------------------------------
+    #
+    def __init__(self, cfg):
+
+        AgentSchedulingComponent.__init__(self, cfg)
+
+    # --------------------------------------------------------------------------
+    #
+    def _configure(self):
+
+        #-----------------------------------------------------------------------
+        # Find out how many applications you can submit to YARN. And also keep
+        # this check happened to update it accordingly
+
+
+        #if 'rm_ip' not in self._cfg['lrms_info']:
+        #    raise RuntimeError('rm_ip not in lm_info for %s' \
+        #            % (self.name))
+
+        self._log.info('Checking rm_ip %s' % self._cfg['lrms_info']['lm_info']['rm_ip'])
+        self._rm_ip = self._cfg['lrms_info']['lm_info']['rm_ip']
+        self._rm_url = self._cfg['lrms_info']['lm_info']['rm_url']
+        self._client_node = self._cfg['lrms_info']['lm_info']['nodename']
+
+        sample_time = rpu.timestamp()
+
+        #-----------------------------------------------------------------------
+        # Find out the cluster's resources
+
+        self._mnum_of_cores = metrics['clusterMetrics']['totalVirtualCores']
+        self._mmem_size = metrics['clusterMetrics']['totalMB']
+        self._num_of_cores = metrics['clusterMetrics']['allocatedVirtualCores']
+        self._mem_size = metrics['clusterMetrics']['allocatedMB']
+
```

Comment:
Ah!  You might want to move `max` then or something.  But wither way, got it, thanks.